<?xml version='1.0'?>
<launch>

	<!-- arguments -->
	<arg name="data" default="test_03"/>


	<!-- replay bag data -->
	<group if="$(eval arg('data')!='real')">

		<param name="use_sim_time" value="true"/>

		<node pkg="rosbag" type="play" name="rosbag_play" args="$(find ugv_localization)/bag/$(arg data).bag --clock" output="screen"/>

	</group>


	<!-- process sensor input messages altogether -->
	<node pkg="ugv_localization" type="topic_process_node" name="topic_process_node" output="screen">
		<rosparam command="load" file="$(find ugv_localization)/params/topic_process.yaml"/>
	</node>	


	<!-- fusion by robot_localization package -->
 	<node pkg="ugv_localization" type="global_filter_wrapper.py" name="global_filter_wrapper" args="/odometry/wheel_imu /odometry/gps /gps/navheading/new" output="screen">
		<param name="filter_node_name" value="ekf_odom_gps_imu"/>
		<param name="topic_out" value="odometry/filtered"/>
		<param name="config_file" value="$(find ugv_localization)/params/global/ekf_odom_imu_gps.yaml"/>
	</node>


    <!-- publish required tf (transformation) -->
	<include file="$(find ugv_localization)/launch/static_tfs.launch"/>
	

	<!-- publish /world/fix and visualize with rviz_satellite -->
	<include file="$(find ugv_localization)/launch/world_frame_gps_publisher.launch"/>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find ugv_localization)/launch/rviz_satellite.rviz"/>


	<!-- visualize map with mapviz -->
	<node pkg="swri_transform_util" type="initialize_origin.py" name="initialize_origin" >
		<param name="local_xy_frame" value="map"/>
    	<param name="local_xy_origin" value="map"/>
    	<rosparam param="local_xy_origins">[{ name: map, latitude: 39.326465, longitude: -76.621554, altitude: 0, heading: 0.0}]</rosparam>
  	</node>

	<node pkg="mapviz" type="mapviz" name="mapviz">
		<param name="config" value="$(find ugv_localization)/launch/mapviz.mvc"/>
	</node>


	<!-- click waypoints and generate trajectory -->
	<include file="$(find ugv_localization)/launch/gen_traj.launch"/>

</launch>
